google/cloud/bigquery/job.py

```diff
@@ -3406,7 +3450,46 @@ def to_dataframe(
         Raises:
             ValueError: If the `pandas` library cannot be imported.
         """
-        return self.result().to_dataframe(
+        query_plan = self.query_plan
+        if query_plan and progress_bar_type:
```

Comment:
`query_plan` can get updated as the job progresses. I'd prefer if this always created a progress bar, but had a generic message when the job is still pending and there isn't a query plan.